scripts/optimize_api.py

```
@@ -0,0 +1,53 @@
+                f"{indent}    foreach ($orders as $order) {{\n",
+                f"{indent}        foreach ($order->get_items() as $item) {{\n",
+                f"{indent}            if ($pid = $item->get_product_id()) {{\n",
+                f"{indent}                $product_ids[] = $pid;\n",
+                f"{indent}            }}\n",
+                f"{indent}        }}\n",
+                f"{indent}    }}\n",
+                f"{indent}    if (!empty($product_ids)) {{\n",
+                f"{indent}        $product_ids = array_unique($product_ids);\n",
+                f"{indent}        update_object_term_cache($product_ids, 'product');\n",
+                f"{indent}    }}\n",
+                f"\n"
```

Ruff (0.15.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)